lib/eventFactory.js

```diff
@@ -2,9 +2,94 @@
 
 const BaseFactory = require('./baseFactory');
 const Event = require('./event');
+const jobConfigSchema = require('screwdriver-data-schema').config.job;
+const joi = require('joi');
 
 let instance;
 
+/**
+ * Create builds associated with this event
+ * For example, if startFrom ~commit, then create builds with jobs that have requires: ~commit
+ * @method createBuilds
+ * @param  {Pipeline} pipeline                   Pipeline to create builds
+ * @param  {Object}   config
+ * @param  {String}   config.sha                 SHA this project was built on
+ * @param  {String}   config.username            Username of the user that creates this event
+ * @param  {String}   config.scmContext          The scm context to which user belongs
+ * @param  {String}   [config.prRef]             Ref if it's a PR event
+ * @param  {String}   [config.prNum]             PR Number if it's a PR event
+ * @param  {String}   [config.startFrom]         Where the event starts from (jobname or ~commit, ~pr, etc)
+ *                                               Making optional for backward compatibility
+ * @param  {Number}   eventId                    Event's id
+ */
+function createBuilds(pipeline, config, eventId) {
+    const startFrom = config.startFrom;
+
+    // If no startFrom, then do nothing. Remove once we switch to new workflow design
+    if (!startFrom) {
+        return null;
```

Comment:
Shouldn't it start build for `main` job ?

Reply:
If it's using the old workflow, we don't want to create the build. Since that means the build will be created twice. For example, this is using the old workflow: https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/builds/create.js#L75-L94

Reply:
Won't this get called for all cases ? Where is the old workflow handling happening ? I thought if it's old workflow then just run `main` else follow `startFrom`

Reply:
We haven't switch everything to the new workflow yet. This is here to ensure backward compatibility. 

If it's using the old design, it will not behave differently from before (aka, just create the event and then done). 

Take a look at the link I included above. It creates the new event with no `startFrom`, so it's using the old design. Hence, it will just create the event, and then it creates the build on its own (https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/builds/create.js#L94)